Studio revamp

proksh opened the pull request:
- Added subfolders in the studio app
> components - contains all the components related to app
> pages - contains main pages only
> api - contains api stuff
> const - contains all the types, constants, and functions that can be used in the app
> storybook - contains storybook components
> style - all the style files

- Revamped the studio design
- Added search components panel
- Added design system
- Changed the structure of the Action modal
- Changed drag-and-drop UI and package for command reorder

Leland-Takamine commented:
Nice work Proksh!

Noticed that the code gets cut off below the action icons on the right:

<img width="637" alt="image" src="https://github.com/mobile-dev-inc/maestro/assets/847683/f3dca9f9-4e68-48e9-8aa5-d24862ee4457">

Leland-Takamine commented on maestro-studio/web/src/pages/InteractPage.tsx:
```diff
@@ -0,0 +1,118 @@
+import ReplView from "../components/commands/ReplView";
+import { DeviceScreen, UIElement } from "../const/models";
+import { API, wait } from "../api/api";
+// import { ActionModal } from "../components/depreciated(toremove)/ActionModal";
```
Comment ^

Leland-Takamine commented on maestro-studio/web/src/pages/InteractPage.tsx:
```diff
@@ -0,0 +1,118 @@
+          setHoveredElement={setHoveredElement}
+          deviceScreen={deviceScreen}
+          onElementSelected={(element: UIElement | null) => {
+            console.log(element);
```
Remove log

Leland-Takamine commented on maestro-studio/web/src/components/commands/CommandRow.tsx:
```diff
@@ -0,0 +1,83 @@
+}: CommandRowProps) {
+  // const [value, setValue] = useState(command.yaml);
+
+  console.log(command.yaml);
```
Remove log

Leland-Takamine commented on maestro-studio/web/src/components/commands/CommandRow.tsx:
```diff
@@ -0,0 +1,83 @@
+  onClick,
+  isDragging,
+}: CommandRowProps) {
+  // const [value, setValue] = useState(command.yaml);
```
Remove comment

Leland-Takamine commented on maestro-studio/web/src/components/commands/CommandRow.tsx:
```diff
@@ -0,0 +1,83 @@
+        <pre className="font-mono cursor-default overflow-x-scroll text-sm text-gray-900 dark:text-white py-0.5 w-full">
+          {command.yaml}
+        </pre>
+        {/* <CommandInput
```
commented code

proksh commented:
@Angelk90 

Updated
- Fixed the browser-related issues you mentioned
- About screen resizing, the app isn't responsive right now, but I did sort the resizing issues till larger tablet devices.

Left out
- Edit an already added field that doesn't have an API right now. It will be updated later
- About the margin on the button, I don't think it's necessary because the button doesn't have a background in the resting state. Let me know if there is any particular reason for mentioning it that I am missing.

proksh commented:
Hey @Angelk90 

I agree with both of your points.

- Regarding the first point, I believe it would be more appropriate to address the issues related to size and responsiveness in a separate pull request at a later stage.

- As for the second point, I share the same sentiment about dark-mode colors. :) Currently, the app is using the default theme provided by Tailwind. Given that implementing changes for dark mode would require significant modifications, it would be more feasible to handle this in a different pull request as well.

Leland-Takamine commented:
Text color issue:
<img width="500" alt="image" src="https://github.com/mobile-dev-inc/maestro/assets/847683/98c1750b-cc44-40bc-89c2-bf0da79e7c86">

axelniklasson reviewed (approved):
Noticed a bug with the export modal in light mode -- aside from that it looks great to me, so approving for now assuming this will be fixed 🙌 

<img width="1425" alt="image" src="https://github.com/mobile-dev-inc/maestro/assets/6113675/35aab3f2-5202-4f07-b242-09c270ced3d0">

proksh commented:
@Angelk90 Nice catch.

Pushed the fix for this.

proksh commented:
@Angelk90 It should be fixed now.
